pertpy/tools/_distance_tests.py

```diff
@@ -0,0 +1,224 @@
+from __future__ import annotations
+
+import numpy as np
+import pandas as pd
+from anndata import AnnData
+from rich.progress import track
+from sklearn.metrics import pairwise_distances
+from statsmodels.stats.multitest import multipletests
+
+from ._distances import Distance
+
+"""
+DEV NOTES:
+- There will be a single PermutationTest class that internally does one of the following strategies:
+    - recomputes the pairwise distance matrix every time and works for
+      any metric that has .
+    - uses the precomputed pairwise distance matrix and works only for
+    metrics that can be called on the pairwise distance matrix. Will be faster.
+
+TODO:
+- Make PermutationTest allow for multiple controls (accept list of controls)
+- Check if PermutationTest works for all metrics
+"""
+
+
+class PermutationTest:
```

Comment:
permutation test is a very general term. Can we make it more explicit that this is related to distances?

Reply:
It sounds general because the framework is very general. But I get your point. How about "DistanceTest"?

Reply:
yup